.github/workflows/test.yml

```diff
@@ -0,0 +1,82 @@
+name: Test
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: write
+  pull-requests: write
+
+concurrency:
+  group: test-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  snapshots:
+    name: Snapshots
+    # Don't run on main
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: '3.0'
+        bundler-cache: true
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 16
+        cache: 'npm'
+        cache-dependency-path: |
+          package-lock.json
+          demo/package-lock.json
+    - run: |
+        npm ci
+        cd demo && npm ci
+    - name: Build and test with Rake
+      run: |
+        bundle exec rake test:snapshots
+    - id: auto-commit
+      uses: stefanzweifel/git-auto-commit-action@v4
+      with:
+        commit_message: Generating component snapshots
+        file_pattern: test/snapshots/**/*.png
+
+    - name: "Changes detected"
+      if: steps.auto-commit.outputs.changes_detected == 'true'
+      uses: phulsechinmay/rewritable-pr-comment@v0.2.1
+      with:
+        message: |
+          ### ⚠️ Visual differences found
+
+          Our visual comparison tests found UI differences. Please review the differences by viewing the files changed tab to ensure that the changes were intentional.
+
+          [Review visual differences](https://github.com/primer/view_components/pull/${{ github.event.number }}/files?file-filters%5B%5D=.png&show-viewed-files=false)
```

Comment:
I linked to the PR tab because we do have those features in GitHub. If there is a change in the image you'll see the swipe and onion skin. You can see it in action on this commit where some images changed. https://github.com/primer/view_components/pull/1437/commits/32002d5aecdfb644e48a21645e0b79e7372627a4